services/wireguard/service.go

```diff
@@ -43,19 +48,27 @@ func NewManager(locationResolver location.Resolver, ipResolver ip.Resolver) *Man
 	}
 }
 
-// Manager represents entrypoint for Wireguard service
+// Manager represents entrypoint for Wireguard service.
 type Manager struct {
+	isStarted        bool
 	process          sync.WaitGroup
 	locationResolver location.Resolver
 	ipResolver       ip.Resolver
-	isStarted        bool
+	wg               Network
 }
 
-// Config represent a Wireguard service provider configuration that will be passed to the consumer for establishing a connection
+// Config represent a Wireguard service provider configuration that will be passed to the consumer for establishing a connection.
 type Config struct {
-	PublicKey string
-	IP        string
-	Endpoint  string
+	Provider network.Provider
+	Consumer network.Consumer
+}
+
+// Network represents Wireguard network instance, it provide information
+// required for establishing connection between service provider and consumer.
+type Network interface {
```

Comment:
Too generic name here. Would be nice to see a hint that it is an interface. I suggest 'Networker'.

Reply:
Renamed to `ConfigProvider` to have clear understandig what it does.